### PR TITLE
A passthrough tokenizer for pre-tokenized datasets

### PR DESCRIPTION
**Pull request by jthickstun**
## Description

Standard GPT-2 models run on text using a pretrained tokenizer. There has been some interest in using Mistral to train models of more data sources (e.g., music) to which the standard GPT-2 tokenizer does not apply. For these settings, it can be convenient to provide the data in the form of _pre-tokenized_ lists of integer tokens.

This pull request defines a PassthroughTokenizer, which treats incoming dataset files as lists of space-separated integers. The PassthroughTokenizer is essentially a no-op, allowing these pre-tokenized values to be passed transparently into the Mistral codebase, skipping the usual tokenization step.

## Known breaking changes/behaviors

The PassthroughTokenizer can be requested by the `passthrough_tokenizer` boolean config flag (default False). There ought to be no changes to the existing Mistral interface.

### CODE REVIEW

File: src/models/auto_clm.py [hunk 1]
```diff
@@ -52,6 +54,9 @@ def get_auto_clm_tokenizer(
     overwatch.info(f"Fetching Hugging Face [Fast] AutoTokenizer for Model: `{REGISTRY[model_id]}`...")
     if use_pretrained_tokenizer:
         tokenizer = AutoTokenizer.from_pretrained(REGISTRY[model_id], config=config, cache_dir=paths["tokenizer"])
+    elif use_passthrough_tokenizer:
```
Comment (dlwh) [inline src/models/auto_clm.py:57]: could we add an assert that at most one of use_pretrained_tokenizer and use_passthrough_tokenizer are set?

**Review by dlwh — APPROVED** [review]
Just one little thing if you don't mind, otherwise lgtm!

File: src/models/auto_clm.py (hunk 1 above)
Comment (jthickstun) [reply src/models/auto_clm.py:58]: Good idea. I added this.

**Comment by dlwh** [comment]
i don't think i closed this???

**Pull request merged**